spec/units/token_parser_spec.rb

```diff
@@ -173,6 +173,17 @@
         env: env, command: 'commit', args: [],
       )
     end
+
+    it 'parses blank lines' do
+      expect(parse(tokens([:EOS]))).to be_a Gitsh::Commands::Noop
+      expect(parse(tokens([:SPACE], [:EOS]))).to be_a Gitsh::Commands::Noop
+    end
+
+    it 'does not parse just a semicolon' do
```

Comment:
Both bash and sh consider `;` to be a syntax error. Combined with the default command considerations you mentioned, I think that's reason enough to keep the current behaviour.